### ISSUE TITLE
Validation Error: Mutable File System | Lesson 7 (Create a Directory)

### ISSUE THREAD

**Issue by zebateira** [ProtoSchool/protoschool.github.io#337]
**URL of the lesson where you received the validation error:** https://proto.school/#/mutable-file-system/07

**The text of the validation error message you saw:** 
"Looks like you edited the ls code to list something other than the root directory. Please try again, editing only the section of code indicated."
![image](https://user-images.githubusercontent.com/2353186/71891974-40fafa00-3161-11ea-959a-8326e0d13ef3.png)

**The code you submitted that caused this error message to appear:**
One line: `await ipfs.files.mkdir('/some/stuff')`
Full Code:
```js
const run = async (files) => {
  // This code adds your uploaded files to your root directory in IPFS
  await Promise.all(files.map(f => ipfs.files.write('/' + f.name, f, { create: true })))

  await ipfs.files.mkdir('/some/stuff')

  let rootDirectoryContents = await ipfs.files.ls('/', { long: true })
  return rootDirectoryContents
}

return run
```

Basically by omitting the `{ parents: true }` option.

**Comment by dominguesgm** [comment]
The problem is in line 81 of the 07.vue lesson file.

We're testing for `result.error.message === 'file does not exist'`, when the error message that shows is similar to

```
Intermediate directory path /ipfs/QmbdVrNA1bdKgs4eL4dBtcg9pFSzdfQAoNGWozSutVBDJB/some does not exist, use the -p flag to create it
```

We should have something like `result.error.message.includes('does not exist, use the -p flag to create it')`

**Comment by terichadbourne** [comment]
Nice catch, @zebateira! I'm assuming IPFS changed their error text recently, since this was previously loading the message correctly for that error. 

@alanshaw Out of curiosity, is there a single place we could go to see all of the error messages IPFS is using, or be notified when changes are made to the text of those messages? Our custom error overrides, which are based on the text of IPFS' numberless errors, will currently break whenever your error text changes.

**Comment by zebateira** [comment]
Another solution would be to use the error code (`ERR_NOT_FOUND` in this instance), because that one if it changes it's a breaking change, whereas error messages can change and are not breaking changes - @alanshaw ?

**Comment by terichadbourne** [comment]
Ah, I may have mistaken prior conversations about IPFS not using numerical error codes for it not using error codes in general. If the error code and the message have a 1-to-1 relationship and this is just a matter of a developer changing the message over time, then I agree with @zebateira.  
 What we need to be careful about is whether the code is delivered in more than one situation (if under the hood IPFS is saying "if Y happened, deliver X error code with Y message; if Z happened, deliver X error code with Z message"), in which case we need to stick to whatever indicator is most situation-specific.

**Comment by dominguesgm** [comment]
Just so the information is in both relevant places, #301 already implements the first approach I mentioned. I think I ran into the issue and added the fix without creating an issue for it.

I think it would be a good idea to close this issue when #301 enters and, if you feel its relevant, open a new issue on improving how specific IPFS errors are detected in the existing lessons' validation code. If we're going to detect them based on error codes, we should probably standardize that approach. What are your thoughts @zebateira?

**Comment by zebateira** [comment]
> open a new issue on improving how specific IPFS errors are detected in the existing lessons' validation code. If we're going to detect them based on error codes, we should probably standardize that approach. What are your thoughts @zebateira?

Sounds good. I'll create a new issue for this discussion 👌

**Comment by zebateira** [comment]
Created here #343
